eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -453,13 +482,13 @@ object Transactions {
                         toLocalDelay: CltvExpiryDelta,
                         localDelayedPaymentPubkey: PublicKey,
                         feeratePerKw: FeeratePerKw,
-                        commitmentFormat: CommitmentFormat): Either[TxGenerationSkipped, HtlcTimeoutTx] = {
+                        commitmentFormat: CommitmentFormat): TxGenerationResult[HtlcTimeoutTx] = {
     val fee = weight2fee(feeratePerKw, commitmentFormat.htlcTimeoutWeight)
     val redeemScript = output.redeemScript
     val htlc = output.commitmentOutput.outgoingHtlc.add
     val amount = htlc.amountMsat.truncateToSatoshi - fee
     if (amount < localDustLimit) {
-      Left(AmountBelowDustLimit)
+      TxGenerationResult.AmountBelowDustLimit
```

Comment:
We can't differentiate between "hard"/"soft" dust limit today. Should we do something like:
```scala
if (amount < HARD_DUST_LIMIT) {
      TxGenerationResult.PermanentDust
} else if (amount < localDustLimit) {
      TxGenerationResult.TemporaryDust
} else {
...
}
```

Sidenote: what would be the relationship between this new `Transactions.HARD_DUST_LIMIT` and the existing `Channel.MIN_DUST_LIMIT`? 🤔